src/main/java/org/terasology/scenario/internal/ui/LogicItemRenderer.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.scenario.internal.ui;
+
+import org.terasology.math.geom.Rect2i;
+import org.terasology.math.geom.Vector2i;
+import org.terasology.rendering.assets.font.Font;
+import org.terasology.rendering.assets.texture.TextureRegion;
+import org.terasology.rendering.nui.Canvas;
+import org.terasology.rendering.nui.TextLineBuilder;
+import org.terasology.rendering.nui.itemRendering.AbstractItemRenderer;
+
+import java.util.List;
+
+public class LogicItemRenderer extends AbstractItemRenderer<LogicTreeValue> {

```

Comment:
This class needs a javadoc explaining why it exists. Looks to me like a copy of an existing one.